src/audio/chain_dma.c

```diff
@@ -333,6 +240,114 @@ static enum task_state chain_task_run(void *data)
 	return SOF_TASK_STATE_RESCHEDULE;
 }
 
+static int chain_task_start(struct comp_dev *dev)
+{
+	struct comp_driver_list *drivers = comp_drivers_get();
+	struct chain_dma_data *cd = comp_get_drvdata(dev);
+	k_spinlock_key_t key;
+	int ret;
+
+	comp_info(dev, "chain_task_start(), host_dma_id = 0x%08x", cd->host_connector_node_id.dw);
+
+	key = k_spin_lock(&drivers->lock);
+	switch (cd->chain_task.state) {
+	case SOF_TASK_STATE_QUEUED:
+		k_spin_unlock(&drivers->lock, key);
+		return 0;
+	case SOF_TASK_STATE_COMPLETED:
+		break;
+	case SOF_TASK_STATE_INIT:
+		break;
+	case SOF_TASK_STATE_FREE:
+		break;
+	default:
+		comp_err(dev, "chain_task_start(), bad state transition");
+		ret = -EINVAL;
+		goto error;
+	}
+
+	if (cd->stream_direction == SOF_IPC_STREAM_PLAYBACK) {
+		ret = chain_host_start(dev);
+		if (ret)
+			goto error;
+		ret = chain_link_start(dev);
+		if (ret) {
+			chain_host_stop(dev);
+			goto error;
+		}
+	} else {
+		ret = chain_link_start(dev);
+		if (ret)
+			goto error;
+		ret = chain_host_start(dev);
+		if (ret) {
+			chain_link_stop(dev);
+			goto error;
+		}
+	}
+
+	pm_policy_state_lock_get(PM_STATE_RUNTIME_IDLE, PM_ALL_SUBSTATES);
```

Comment:
I can not find a symmetric pm_policy_state_lock_put() from the successful code path. Is it handled somewhere out side this function, or is it missing?

Reply:
`pm_policy_state_lock_put()` symmetric from successful code path can be found in at the end of `chain_task_pause` in under label called `error:`.